mono/mini/mini-amd64.c

```diff
@@ -1195,16 +1212,33 @@ mono_arch_set_native_call_context_args (CallContext *ccontext, gpointer frame, M
 
 	for (int i = 0; i < sig->param_count; i++) {
 		ainfo = &cinfo->args [i];
-		int temp_size = arg_need_temp (ainfo);
 
-		if (temp_size)
-			storage = alloca (temp_size); // FIXME? alloca in a loop
-		else
-			storage = arg_get_storage (ccontext, ainfo);
+		switch (ainfo->storage) {
+			case ArgInIReg:
+			case ArgInFloatSSEReg:
+			case ArgInDoubleSSEReg:
+			case ArgOnStack:
+			case ArgValuetypeInReg: {
+				int temp_size = arg_need_temp (ainfo);
+				if (temp_size)
+					storage = alloca (temp_size); // FIXME? alloca in a loop
+				else
+					storage = arg_get_storage (ccontext, ainfo);
 
-		interp_cb->frame_arg_to_data ((MonoInterpFrameHandle)frame, sig, i, storage);
-		if (temp_size)
-			arg_set_val (ccontext, ainfo, storage);
+				interp_cb->frame_arg_to_data ((MonoInterpFrameHandle)frame, sig, i, storage);
+				if (temp_size)
+					arg_set_val (ccontext, ainfo, storage);
+				break;
+			}
+			case ArgValuetypeAddrInIReg:
+			case ArgValuetypeAddrOnStack: {
+				storage = arg_get_storage (ccontext, ainfo);
+				*(gpointer *)storage = interp_cb->frame_arg_to_storage (frame, sig, i);
+				break;
```

Comment:
Could we maybe do it in the same fashion as on arm64 (by avoiding big switch and less diff) ?

Reply:
ok